programs/ido-pool/src/lib.rs

```diff
@@ -162,14 +198,22 @@ pub mod ido_pool {
             to: ctx.accounts.user_usdc.to_account_info(),
             authority: ctx.accounts.pool_signer.to_account_info(),
         };
-        let cpi_program = ctx.accounts.token_program.clone();
+        let cpi_program = ctx.accounts.token_program.to_account_info();
         let cpi_ctx = CpiContext::new_with_signer(cpi_program, cpi_accounts, signer);
         token::transfer(cpi_ctx, amount)?;
 
+        // Update USDC amounts on the pool account
+        ctx.accounts.pool_account.num_usdc_tokens = (ctx.accounts.pool_account.num_usdc_tokens
+            as u128)
+            .checked_sub(amount as u128)
+            .unwrap()
+            .try_into()
+            .unwrap();
```

Comment:
same comment as https://github.com/igneous-labs/ido-pool/pull/1/files#r772047277, is there a need to cast into u128 if it's just subtraction?